src/Libraries/Nop.Services/Helpers/BrowscapXmlParser.cs

```diff
@@ -101,7 +104,11 @@ private static string ToRegexp(string str)
         /// <returns>True if user agent is a crawler, otherwise - false</returns>
         public bool IsCrawler(string userAgent)
         {
-            return _crawlerUserAgentsRegexp.Any(p => Regex.IsMatch(userAgent, p));
+            // check against cache object
+            bool result;
+            if (_userAgentsCache.TryGetValue(userAgent, out result))
```

Comment:
Will look into, originally had request ua's and their result stored, may need to do discovery on the performance of the permutations but likely still significantly faster than the regex.